src/LinearMap.jl

```diff
@@ -21,106 +21,169 @@ independent of the numeric type of the wrapped set (`N`).
+LinearMap{Int64,BallInf{Int64},Int64,SparseArrays.SparseMatrixCSC{Int64,Int64}}(
+  [1, 1]  =  2
+  [2, 2]  =  2, BallInf{Int64}([0, 0], 1))
+
+julia> 2I * X == lm3
+true
+
+julia> 1I * X == X
+true
+```
+
+Applying a linear map to a `LinearMap` object combines the two maps into a
+single `LinearMap` instance.
+Again we can make use of the conversion for convenience.
+
+```jldoctest constructors
+julia> B = transpose(A); B * lm
+LinearMap{Int64,BallInf{Int64},Int64,Array{Int64,2}}([3 9; 9 29], BallInf{Int64}([0, 0], 1))
+
+julia> B = [3, 4, 5]; B * lm
+LinearMap{Int64,BallInf{Int64},Int64,Array{Int64,2}}([12 38], BallInf{Int64}([0, 0], 1))
+
+julia> B = 2; B * lm
+LinearMap{Int64,BallInf{Int64},Int64,Array{Int64,2}}([2 4; 2 6; 2 8], BallInf{Int64}([0, 0], 1))
+```
+
+The application of a `LinearMap` to a `ZeroSet` or an `EmptySet` is simplified
+automatically.
+
+```jldoctest constructors
+julia> A * ZeroSet{Int}(2)
+ZeroSet{Int64}(3)
+
+julia> A * EmptySet{Int}()
+EmptySet{Int64}()
+```
 """
 struct LinearMap{N<:Real, S<:LazySet{N},
                  NM, MAT<:AbstractMatrix{NM}} <: LazySet{N}
     M::MAT
     X::S
 
     # default constructor with dimension match check
-    function LinearMap{N, S, NM, MAT}(M::MAT, X::S) where {N<:Real,
-            S<:LazySet{N}, NM, MAT<:AbstractMatrix{NM}}
+    function LinearMap(M::MAT, X::S) where {N<:Real, S<:LazySet{N}, NM,
+                                            MAT<:AbstractMatrix{NM}}
         @assert dim(X) == size(M, 2) "a linear map of size $(size(M)) cannot " *
             "be applied to a set of dimension $(dim(X))"
         return new{N, S, NM, MAT}(M, X)
     end
 end
 
-# convenience constructor without type parameter
-LinearMap(M::MAT,
-          X::S) where {N<:Real, S<:LazySet{N}, NM, MAT<:AbstractMatrix{NM}} =
-    LinearMap{N, S, NM, MAT}(M, X)
-
-# constructor from a linear map: perform the matrix multiplication immediately
-LinearMap(M::MAT, lm::LinearMap{N, S, NM, MAT}) where {N<:Real, S<:LazySet{N},
-        NM, MAT<:AbstractMatrix{NM}} =
-    LinearMap{N, S, NM, MAT}(M * lm.M, lm.X)
-
 """
 ```
-    *(M::AbstractMatrix{N}, X::LazySet{N}) where {N<:Real}
+    *(map::Union{AbstractMatrix, UniformScaling, AbstractVector, Real}, X::LazySet)
 ```
 
-Return the linear map of a convex set.
+Alias to create a `LinearMap` object.
 
 ### Input
 
-- `M` -- matrix/linear map
-- `X` -- convex set
+- `map` -- linear map
+- `X`   -- convex set
 
 ### Output
 
-A lazy linear map, i.e. a `LinearMap` instance.
+A lazy linear map, i.e., a `LinearMap` instance.
 """
-function *(M::AbstractMatrix{N}, X::LazySet{N}) where {N<:Real}
-    return LinearMap(M, X)
+function *(map::Union{AbstractMatrix, UniformScaling, AbstractVector, Real}, X::LazySet)
+    return LinearMap(map, X)
 end
 
-function *(M::AbstractVector{N}, X::LazySet{N}) where {N<:Real}
-    return LinearMap(reshape(M, length(M), 1), X)
+# convenience constructor from a vector
+function LinearMap(v::AbstractVector, X::LazySet)
```

Comment:
we can keep it, but tthinking again of this constructor, it is admittedly rather strange given that `v` is a *vector* so the interpretation of `vX` could vary; i dont remember why we added it in the first place.

Reply:
If you do not like it anymore, I suggest to remove it later to keep the changes in this PR limited.